### PR TITLE
Add Oracle SQL - Insert statement

### PR DESCRIPTION
**Pull request by ThanoshanMV**
#10111

Hi @wgy8283335. I've added SQL definition for Oracle `INSERT` statement. Please check it.

Changes proposed in this pull request:
- Add Oracle [INSERT ](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/INSERT.html#GUID-903F8043-0254-4EE9-ACC1-CB8AC0AF3423) statement.
- Add corresponding test cases.

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4 [hunk 1]
```diff
@@ -48,11 +48,45 @@ conditionalInsertElsePart
...
+dmlSubqueryClause
+    : LP_ selectSubquery subqueryRestrictionClause? RP_
+    ;
+
+subqueryRestrictionClause
+    : WITH (READ ONLY | CHECK OPTION) (CONSTRAINT constraintName)?
+    ;
+
+tableCollectionExpr
+    : TABLE LP_ collectionExpr RP_ (LP_ PLUS_ RP_)?
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4:85]: Where is the definition of collection expression in ORALCE site?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dml/OracleInsertStatement.java [hunk 2]
```diff
@@ -17,13 +17,41 @@
...
+import java.util.Optional;
+
 /**
  * Oracle insert statement.
  */
+@Setter
 @ToString
 public final class OracleInsertStatement extends InsertStatement implements OracleStatement {
+    
+    private SubquerySegment selectSubquery;
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dml/OracleInsertStatement.java:36]: Why add selectSubquery and insertMultiTableElementSegment?

Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dml/OracleInsertStatement.java:36]: In an `INSERT` statement, with the help of `insertIntoClause` we can add table elements such as `SimpleTableSegment`, `SubquerySegment`. Those are available in `InsertStatement`. But in Oracle's `insertSingleTable` and `insertMultiTable` clause, it expects a `SubquerySegment` after `insertIntoClause`. I think this is specific to Oracle insert statement.

insertSingleTable

insertMultiTable

Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/oracle/dml/OracleInsertStatement.java:36]: I added `insertMultiTableElementSegment` to support multitable inserts.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4 (hunk 1 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4:85]: Here's the definition:
tableCollectionExpression

**Comment by tristaZero** [comment]
@ThanoshanMV Hi 
Looks #11000 caused some conflicts, could you have a look?

@wgy8283335 Are there any improvements to do for this PR?

**Review by wgy8283335 — APPROVED** [review]

**Comment by tristaZero** [comment]
Hi guys, 

I guess you missed [this post](https://mp.weixin.qq.com/s/hqFHEAPZHcoEfC4ghj3QSA), which introduced the features of 5.0.0-beta release and still presented all the contributors for this release. Again, thanks for your effort in Apache ShardingSphere community.

**Pull request merged**